legacy/src/main/java/org/opensearch/sql/legacy/executor/join/ElasticJoinExecutor.java

```diff
@@ -256,23 +280,47 @@ protected void updateMetaSearchResults(SearchResponse searchResponse) {
     this.metaResults.updateTimeOut(searchResponse.isTimedOut());
   }
 
-  protected SearchResponse scrollOneTimeWithMax(
-      Client client, TableInJoinRequestBuilder tableRequest) {
-    SearchRequestBuilder scrollRequest =
-        tableRequest
-            .getRequestBuilder()
-            .setScroll(new TimeValue(60000))
-            .setSize(MAX_RESULTS_ON_ONE_FETCH);
-    boolean ordered = tableRequest.getOriginalSelect().isOrderdSelect();
-    if (!ordered) {
-      scrollRequest.addSort(FieldSortBuilder.DOC_FIELD_NAME, SortOrder.ASC);
+  public SearchResponse getResponseWithHits(
+      TableInJoinRequestBuilder tableRequest, int size, SearchResponse previousResponse) {
+    // Set Size
+    SearchRequestBuilder request = tableRequest.getRequestBuilder().setSize(size);
+    SearchResponse responseWithHits;
+    if (LocalClusterState.state().getSettingValue(SQL_PAGINATION_API_SEARCH_AFTER)) {
+      // Set sort field for search_after
+      boolean ordered = tableRequest.getOriginalSelect().isOrderdSelect();
+      if (!ordered) {
+        request.addSort(FieldSortBuilder.DOC_FIELD_NAME, SortOrder.ASC);
+      }
+      // Set PIT
+      request.setPointInTime(new PointInTimeBuilder(pit.getPitId()));
```

Comment:
what if pit request didn't complete by the time code execution reaches this line?

Reply:
As commented in PointInTimeHandler, we are actually listening to the response in listener instead of waiting.

Reply:
Tried printing current thread ID in run and while getting id from pitResponse, thread id is same so it's waiting. Looks like pit request will complete.

Reply:
Ok. Strange if it is working..I am good but I am still not sure...can you add logs somewhere.

Reply:
Added logs in pit handler impl

Reply:
I actually mean thread id logs to comment or description. I am fine anyways.